src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -125,44 +143,47 @@ namespace OpenMS
 
 
   IdentificationData::ProcessingStepRef
-  IdentificationData::registerDataProcessingStep(
-    const DataProcessingStep& step)
+  IdentificationData::registerProcessingStep(
+    const ProcessingStep& step)
   {
-    return registerDataProcessingStep(step, db_search_params_.end());
+    return registerProcessingStep(step, db_search_params_.end());
   }
 
 
   IdentificationData::ProcessingStepRef
-  IdentificationData::registerDataProcessingStep(
-    const DataProcessingStep& step, SearchParamRef search_ref)
+  IdentificationData::registerProcessingStep(
+    const ProcessingStep& step, SearchParamRef search_ref)
   {
-    // valid reference to software is required:
-    if (!isValidReference_(step.software_ref, processing_softwares_))
+    if (!no_checks_)
```

Comment:
minor: I would have preferred if (check_inserts_) as opposed to the double negation

Reply:
It's this way around to have a flag that's off by default, i.e. you have to explicitly enable "no checks".